mpmp/scripts/subsample_data.py

```diff
@@ -0,0 +1,76 @@
+"""
+Generate (or regenerate) subsampled datasets for debugging changes.
+"""
+import argparse
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+import mpmp.config as cfg
+import mpmp.utilities.data_utilities as du
+
+def subsample_stratified(data_df, sample_info_df):
+    """Subsample from data_df, stratified by id in sample_info_df."""
+
+    # reindex sample info to samples found in data
+    stratify_samples_id = (
+        sample_info_df.reindex(data_df.index)
+                      .dropna()
```

Comment:
Yeah, this is intentional. The only case when there would be a NA value here would be if the `sample_info_df.index` and `data_df.index` aren't the same, since I'm filtering out other NA rows in my data download script.

Whether or not I should be filtering all NA rows in my download script is a separate question, and probably a question that I should think more about (particularly in the case of methylation data where some probes/CpG locations aren't measured in all samples). Created #5 to think more about this.